### ISSUE TITLE
[REVIEW]: starfish: scalable pipelines for image-based transcriptomics

### ISSUE THREAD

**Issue by whedon** [openjournals/joss-reviews#2440]
**Submitting author:** @neuromusic (<a href="http://orcid.org/0000-0002-4638-7015">Justin Kiggins</a>)
**Repository:** <a href="https://github.com/spacetx/starfish" target ="_blank">https://github.com/spacetx/starfish</a>
**Version:** 0.2.2
**Editor:** @csoneson
**Reviewer:** @giovp, @shazanfar, @vals
**Archive:** <a href="https://doi.org/10.5281/zenodo.4734993" target="_blank">10.5281/zenodo.4734993</a>

**:warning: JOSS reduced service mode :warning:**

Due to the challenges of the COVID-19 pandemic, JOSS is currently operating in a "reduced service mode". You can read more about what that means in [our blog post](https://blog.joss.theoj.org/2020/05/reopening-joss).

## Status

[![status](https://joss.theoj.org/papers/e00429a8b9030f0a231c870f9602ea7d/status.svg)](https://joss.theoj.org/papers/e00429a8b9030f0a231c870f9602ea7d)

Status badge code:

```
HTML: <a href="https://joss.theoj.org/papers/e00429a8b9030f0a231c870f9602ea7d"><img src="https://joss.theoj.org/papers/e00429a8b9030f0a231c870f9602ea7d/status.svg"></a>
Markdown: [![status](https://joss.theoj.org/papers/e00429a8b9030f0a231c870f9602ea7d/status.svg)](https://joss.theoj.org/papers/e00429a8b9030f0a231c870f9602ea7d)
```
**Reviewers and authors**:

Please avoid lengthy details of difficulties in the review thread. Instead, please create a new issue in the <a href="https://github.com/spacetx/starfish" target="_blank">target repository</a> and link to those issues (especially acceptance-blockers) by leaving comments in the review thread below. (For completists: if the target issue tracker is also on GitHub, linking the review thread in the issue or vice versa will create corresponding breadcrumb trails in the link target.)

## Reviewer instructions & questions

@giovp & @shazanfar & @vals, please carry out your review in this issue by updating the checklist below. If you cannot edit the checklist please:
1. Make sure you're logged in to your GitHub account
2. Be sure to accept the invite at this URL: https://github.com/openjournals/joss-reviews/invitations

The reviewer guidelines are available here: https://joss.readthedocs.io/en/latest/reviewer_guidelines.html. Any questions/concerns please let @csoneson know.

✨ **Please try and complete your review in the next six weeks** ✨ 

### Conflict of interest

- [x] I confirm that I have read the [JOSS conflict of interest (COI) policy](https://github.com/openjournals/joss/blob/master/COI.md) and that: I have no COIs with reviewing this work or that any perceived COIs have been waived by JOSS for the purpose of this review.

### Code of Conduct

- [x] I confirm that I read and will adhere to the [JOSS code of conduct](https://joss.theoj.org/about#code_of_conduct).

### General checks

- [x] **Repository:** Is the source code for this software available at the <a target="_blank" href="https://github.com/spacetx/starfish">repository url</a>?
- [x] **License:** Does the repository contain a plain-text LICENSE file with the contents of an [OSI approved](https://opensource.org/licenses/alphabetical) software license?
- [x] **Contribution and authorship:** Has the submitting author (@neuromusic) made major contributions to the software? Does the full list of paper authors seem appropriate and complete?

### Functionality

- [x] **Installation:** Does installation proceed as outlined in the documentation?
- [x] **Functionality:** Have the functional claims of the software been confirmed?
- [x] **Performance:** If there are any performance claims of the software, have they been confirmed? (If there are no claims, please check off this item.)

### Documentation

- [x] **A statement of need:** Do the authors clearly state what problems the software is designed to solve and who the target audience is?
- [x] **Installation instructions:** Is there a clearly-stated list of dependencies? Ideally these should be handled with an automated package management solution.
- [x] **Example usage:** Do the authors include examples of how to use the software (ideally to solve real-world analysis problems).
- [x] **Functionality documentation:** Is the core functionality of the software documented to a satisfactory level (e.g., API method documentation)?
- [x] **Automated tests:** Are there automated tests or manual steps described so that the functionality of the software can be verified?
- [x] **Community guidelines:** Are there clear guidelines for third parties wishing to 1) Contribute to the software 2) Report issues or problems with the software 3) Seek support

### Software paper

- [x] **Summary:** Has a clear description of the high-level functionality and purpose of the software for a diverse, non-specialist audience been provided?
- [x] **A statement of need:** Do the authors clearly state what problems the software is designed to solve and who the target audience is?
- [x] **State of the field:** Do the authors describe how this software compares to other commonly-used packages?
- [x] **Quality of writing:** Is the paper well written (i.e., it does not require editing for structure, language, or writing quality)?
- [x] **References:** Is the list of references complete, and is everything cited appropriately that should be cited (e.g., papers, datasets, software)? Do references in the text use the proper [citation syntax]( https://rmarkdown.rstudio.com/authoring_bibliographies_and_citations.html#citation_syntax)?

### Conflict of interest

- [x] I confirm that I have read the [JOSS conflict of interest (COI) policy](https://github.com/openjournals/joss/blob/master/COI.md) and that: I have no COIs with reviewing this work or that any perceived COIs have been waived by JOSS for the purpose of this review.

### Code of Conduct

- [x] I confirm that I read and will adhere to the [JOSS code of conduct](https://joss.theoj.org/about#code_of_conduct).

### General checks

- [x] **Repository:** Is the source code for this software available at the <a target="_blank" href="https://github.com/spacetx/starfish">repository url</a>?
- [x] **License:** Does the repository contain a plain-text LICENSE file with the contents of an [OSI approved](https://opensource.org/licenses/alphabetical) software license?
- [x] **Contribution and authorship:** Has the submitting author (@neuromusic) made major contributions to the software? Does the full list of paper authors seem appropriate and complete?

### Functionality

- [x] **Installation:** Does installation proceed as outlined in the documentation?
- [x] **Functionality:** Have the functional claims of the software been confirmed?
- [x] **Performance:** If there are any performance claims of the software, have they been confirmed? (If there are no claims, please check off this item.)

### Documentation

- [x] **A statement of need:** Do the authors clearly state what problems the software is designed to solve and who the target audience is?
- [x] **Installation instructions:** Is there a clearly-stated list of dependencies? Ideally these should be handled with an automated package management solution.
- [x] **Example usage:** Do the authors include examples of how to use the software (ideally to solve real-world analysis problems).
- [x] **Functionality documentation:** Is the core functionality of the software documented to a satisfactory level (e.g., API method documentation)?
- [x] **Automated tests:** Are there automated tests or manual steps described so that the functionality of the software can be verified?
- [x] **Community guidelines:** Are there clear guidelines for third parties wishing to 1) Contribute to the software 2) Report issues or problems with the software 3) Seek support

### Software paper

- [x] **Summary:** Has a clear description of the high-level functionality and purpose of the software for a diverse, non-specialist audience been provided?
- [x] **A statement of need:** Do the authors clearly state what problems the software is designed to solve and who the target audience is?
- [x] **State of the field:** Do the authors describe how this software compares to other commonly-used packages?
- [x] **Quality of writing:** Is the paper well written (i.e., it does not require editing for structure, language, or writing quality)?
- [x] **References:** Is the list of references complete, and is everything cited appropriately that should be cited (e.g., papers, datasets, software)? Do references in the text use the proper [citation syntax]( https://rmarkdown.rstudio.com/authoring_bibliographies_and_citations.html#citation_syntax)?

### Conflict of interest

- [x] I confirm that I have read the [JOSS conflict of interest (COI) policy](https://github.com/openjournals/joss/blob/master/COI.md) and that: I have no COIs with reviewing this work or that any perceived COIs have been waived by JOSS for the purpose of this review.

### Code of Conduct

- [x] I confirm that I read and will adhere to the [JOSS code of conduct](https://joss.theoj.org/about#code_of_conduct).

### General checks

- [x] **Repository:** Is the source code for this software available at the <a target="_blank" href="https://github.com/spacetx/starfish">repository url</a>?
- [x] **License:** Does the repository contain a plain-text LICENSE file with the contents of an [OSI approved](https://opensource.org/licenses/alphabetical) software license?
- [x] **Contribution and authorship:** Has the submitting author (@neuromusic) made major contributions to the software? Does the full list of paper authors seem appropriate and complete?

### Functionality

- [x] **Installation:** Does installation proceed as outlined in the documentation?
- [x] **Functionality:** Have the functional claims of the software been confirmed?
- [x] **Performance:** If there are any performance claims of the software, have they been confirmed? (If there are no claims, please check off this item.)

### Documentation

- [x] **A statement of need:** Do the authors clearly state what problems the software is designed to solve and who the target audience is?
- [x] **Installation instructions:** Is there a clearly-stated list of dependencies? Ideally these should be handled with an automated package management solution.
- [x] **Example usage:** Do the authors include examples of how to use the software (ideally to solve real-world analysis problems).
- [x] **Functionality documentation:** Is the core functionality of the software documented to a satisfactory level (e.g., API method documentation)?
- [x] **Automated tests:** Are there automated tests or manual steps described so that the functionality of the software can be verified?
- [x] **Community guidelines:** Are there clear guidelines for third parties wishing to 1) Contribute to the software 2) Report issues or problems with the software 3) Seek support

### Software paper

- [x] **Summary:** Has a clear description of the high-level functionality and purpose of the software for a diverse, non-specialist audience been provided?
- [x] **A statement of need:** Do the authors clearly state what problems the software is designed to solve and who the target audience is?
- [x] **State of the field:** Do the authors describe how this software compares to other commonly-used packages?
- [x] **Quality of writing:** Is the paper well written (i.e., it does not require editing for structure, language, or writing quality)?
- [x] **References:** Is the list of references complete, and is everything cited appropriately that should be cited (e.g., papers, datasets, software)? Do references in the text use the proper [citation syntax]( https://rmarkdown.rstudio.com/authoring_bibliographies_and_citations.html#citation_syntax)?

**Comment by whedon** [comment]
Hello human, I'm @whedon, a robot that can help you with some common editorial tasks. @giovp, @shazanfar, @vals it looks like you're currently assigned to review this paper :tada:.

**:warning: JOSS reduced service mode :warning:**

Due to the challenges of the COVID-19 pandemic, JOSS is currently operating in a "reduced service mode". You can read more about what that means in [our blog post](https://blog.joss.theoj.org/2020/05/reopening-joss).

**:star: Important :star:**

If you haven't already, you should seriously consider unsubscribing from GitHub notifications for this (https://github.com/openjournals/joss-reviews) repository. As a reviewer, you're probably currently watching this repository which means for GitHub's default behaviour you will receive notifications (emails) for all reviews 😿

To fix this do the following two things:

1. Set yourself as 'Not watching' https://github.com/openjournals/joss-reviews:

![watching](https://cloud.githubusercontent.com/assets/4483/20250593/64d7ce48-a9de-11e6-9225-d3dfb3e48e68.png)

2. You may also like to change your default settings for this watching repositories in your GitHub profile here: https://github.com/settings/notifications

![notifications](https://cloud.githubusercontent.com/assets/4483/20250604/79eb3144-a9de-11e6-9e51-0fe940966689.png)


For a list of things I can do to help you, just type:

```
@whedon commands
```

For example, to regenerate the paper pdf after making changes in the paper's md or bib files, type:

```
@whedon generate pdf
```

**Comment by whedon** [comment]
```
Reference check summary:

OK DOIs

- 10.1016/j.cell.2019.05.031 is OK
- 10.1016/j.cell.2018.05.035 is OK
- 10.1038/nmeth.3252 is OK
- 10.1186/s13059-017-1382-0 is OK
- 10.1073/pnas.1612826113 is OK
- 10.1038/nmeth.2563 is OK
- 10.1038/s41592-018-0175-z is OK
- 10.1093/nar/gkx1206 is OK
- 10.1126/science.aat5691 is OK
- 10.5281/zenodo.59499 is OK
- 10.7717/peerj.453 is OK
- 10.5281/zenodo.3710410 is OK

MISSING DOIs

- None

INVALID DOIs

- None
```

**Comment by whedon** [comment]
[ :point_right: Check article proof :page_facing_up: :point_left: ](https://github.com/openjournals/joss-papers/blob/joss.02440/joss.02440/10.21105.joss.02440.pdf)

**Comment by csoneson** [comment]
👋 @giovp, @shazanfar, @vals - thanks for accepting to review this submission! This is where the review will take place. In the post above you can find your respective checklists, and some additional information and instructions. Don't hesitate to ping me if you have any questions!

**Comment by csoneson** [comment]
@whedon re-invite @vals as reviewer

**Comment by whedon** [comment]
OK, the reviewer has been re-invited.

@vals please accept the invite by clicking this link: https://github.com/openjournals/joss-reviews/invitations

**Comment by giovp** [comment]
To me, first round of review finished, would like to see spacetx/starfish#1899 addressed, in order to explore results.

**Comment by vals** [comment]
First round finished from my end as well. I would like all my issues addressed. But most importantly from my end there should be versions of the practical examples that can be read without running the notebooks ( https://github.com/spacetx/starfish/issues/1904 ). Even when napari is used, the authors could make versions with screenshots. There is a lot of documentation about all the functionality, but without examples it is hard to get an understanding what the different functionality does.

**Comment by csoneson** [comment]
Thanks @giovp and @vals for your reviews! @shazanfar - could you give a quick update on the status on your side? Thanks!

**Comment by shazanfar** [comment]
Thanks for your patience @csoneson ! I've now completed the first round. I agree with @giovp and @vals comments regarding example data and including output in the tutorials. I've also requested amendments & clarification to do with scope + target audience, which I think would make the documentation more easily accessible for users. Thanks again! Cheers, Shila

**Comment by csoneson** [comment]
Thanks @shazanfar! 
@neuromusic - looks like you got some good comments from the reviewers (and that you already started working on some of them 😃). Let me know if you have any questions, and when you are ready for the reviewers to take another look. Thanks!

**Comment by neuromusic** [comment]
Thanks all! @mattcai and I have started addressing a few of your comments and issues and we will follow up on the rest in the coming weeks.

**Comment by neuromusic** [comment]
Quick question: How would you prefer we handle closing issues when we feel like we've addressed them? Should we close them or is it best to wait for our reviewers to close them when they are happy?

**Comment by csoneson** [comment]
@neuromusic - if you are confident that the issue has been addressed I think you can close the issue, with a comment explaining the changes that were made. You can also consult with the reviewer first, in case additional clarifications are needed.

**Comment by csoneson** [comment]
👋 @neuromusic - just wanted to check in quickly to see how things are going. Let us know if there are any questions.

**Comment by csoneson** [comment]
ping @neuromusic 
>👋 @neuromusic - just wanted to check in quickly to see how things are going. Let us know if there are any questions.

**Comment by neuromusic** [comment]
Thanks for the ping :D I'll follow up on the remaining issues shortly.

**Comment by csoneson** [comment]
👋 @neuromusic - just checking in here. Do you have an ETA for the remaining fixes? Of course, let us know if there are any questions.

**Comment by neuromusic** [comment]
sigh. time flies. I'll take care of the remaining issues this week.

**Comment by csoneson** [comment]
> sigh. time flies. I'll take care of the remaining issues this week.
> 
> sorry about the delay :(

@neuromusic - any update on this?

**Comment by csoneson** [comment]
@neuromusic - any update on the remaining issues?

**Comment by csoneson** [comment]
@neuromusic, please update us on the status of the remaining issues. If you don't have time to work on this now, let us know and we'll pause this submission.

**Comment by csoneson** [comment]
I will pause this submission for now. 
@neuromusic - do you think you will have the time to work on the remaining issues within the next month? If not, in order to respect the reviewers' time and availability committment, I will propose to close (reject) this submission, and invite a resubmission at a later point in time.

**Comment by neuromusic** [comment]
I'm sorry for my poor communication over the past month. I've been up against some deadlines, but I expect to be able to address the remaining issues within the next 3 weeks.

**Comment by csoneson** [comment]
Thanks @neuromusic!

**Comment by csoneson** [comment]
👋 @neuromusic - just checking in that we're still on track to have these issues addressed within the next week.

**Comment by csoneson** [comment]
> 👋 @neuromusic - just checking in that we're still on track to have these issues addressed within the next week.

ping @neuromusic - could you please give us an update on your progress?

**Comment by neuromusic** [comment]
you can expect a response to the outstanding issues tomorrow :D

**Comment by neuromusic** [comment]
Thank you for your patience, @csoneson and all.

I've addressed the outstanding issues from @vals and @shazanfar in [PR #1913](https://github.com/spacetx/starfish/pull/1913), except for the request for rendered examples in [starfish#1904](https://github.com/spacetx/starfish/issues/1904), which I've addressed in a comment on that issue.

As soon as the tests pass and my colleague @mattcai reviews the PR, these changes will be live in the manuscript and documentation.

**Comment by csoneson** [comment]
Thanks @neuromusic. I have removed the `paused` label. Once the changes are merged you can ask whedon to re-render the manuscript: `@whedon generate pdf`, and @giovp, @shazanfar, @vals can have a look at the revisions.

**Comment by neuromusic** [comment]
@whedon generate pdf

**Comment by whedon** [comment]
:point_right::page_facing_up: [Download article proof](https://raw.githubusercontent.com/openjournals/joss-papers/joss.02440/joss.02440/10.21105.joss.02440.pdf) :page_facing_up: [View article proof on GitHub](https://github.com/openjournals/joss-papers/blob/joss.02440/joss.02440/10.21105.joss.02440.pdf) :page_facing_up: :point_left:

**Comment by neuromusic** [comment]
Heads up @csoneson we're going to need to make a couple of tweaks to the Acknowledgements before this is published. We should have them by the end of the week.

**Comment by csoneson** [comment]
Sure, no problem. The next step now is for the reviewers to have another look at the submission and the changes that you have made, and update their checklists above accordingly if they're happy with the current state. 

@giovp, @shazanfar, @vals - this should be ready for another look from you now. Don't hesitate to let me know if there are any questions. Thanks!

**Comment by shazanfar** [comment]
Thanks @csoneson and @neuromusic ! I have read through the new article and the updated software documentation and amended my reviewing checklist accordingly. Cheers, Shila

**Comment by giovp** [comment]
Hi @csoneson and @neuromusic , also went through the manuscript again and updated the checklist !

**Comment by csoneson** [comment]
Thanks @shazanfar and @giovp! 

@vals - would you also still be willing to check the updated version of this submission? Thanks!

**Comment by vals** [comment]
Hi @csoneson yes I'll get to it shortly

**Comment by vals** [comment]
Hi again @csoneson , I have gone over the checklist again now for a second round.

There are still some issues regarding the documentation (linked above). Additionally, the manuscript doesn't mention how starfish differ from the alternative tools mentioned. I'm sure the authors could write some comments on how flexible and scalable starfish is compared to alternative tools.

Best,
/Valentine

**Comment by csoneson** [comment]
Thanks @vals - @neuromusic, could you take a look at expanding the documentation a bit and highlighting how starfish compares to other tools, as indicated above?

**Comment by neuromusic** [comment]
I think we can address most of them, however, I need to look into https://github.com/spacetx/starfish/issues/1925 more carefully... I've explored automatically capturing napari screenshots, but I'm unsure of how easy it will be to do as part of our documentation build process.

**Comment by vals** [comment]
Do they need to be captured automatically? There are other screenshots in the tutorials. The examples I gave are actually exceptions. It is hard to follow what is going on with the Napari window if it's not pointed out with images in the documentation.

**Comment by neuromusic** [comment]
@vals yes, its important to us that all documentation involving data can be automatically generated so that the images don't become stale if the tutorial gets updated in the future. all of the images in the other tutorials are generated through matplotlib, etc. 

let me look into this a bit more. I've got a PR open where I started trying to capture napari screenshots programmatically, which I need to revisit. It also might make sense to simply use matplotlib for the quick start instead of napari

**Comment by csoneson** [comment]
👋 Just wanted to check in on the status here before the holidays.

**Comment by csoneson** [comment]
Happy New Year! @neuromusic - could you give an update of where we are standing wrt the remaining issues? Thanks!

**Comment by csoneson** [comment]
Just checking in again to see where we are at here (@neuromusic).

**Comment by csoneson** [comment]
@neuromusic - please give us an update on where things are and what the plan to address the remaining few issues.

**Comment by csoneson** [comment]
@neuromusic - please let us know how this is progressing. It looks like we are pretty close to the finishing line, but at this point I would like to ask for a timeline of when you expect to be able to address the remaining comments, or whether there are issues that need to be discussed.

**Comment by neuromusic** [comment]
Thanks, I'll provide an update on the outstanding item shortly.

**Comment by neuromusic** [comment]
quick update: I've got screenshot rendering working locally. I just need to get it working on CI & readthedocs. Hoping to get that taken care of this week.

**Comment by csoneson** [comment]
@neuromusic any updates?

**Comment by csoneson** [comment]
@neuromusic - please keep us updated on the progress here, and provide a more precise timeline that you can commit to for addressing the few outstanding items. I'm aware that these are trying times for many of us. However, to respect the reviewers' commitment, I want to make sure that the submissions are actively worked on. If this is not possible at the moment, as mentioned last year, I would suggest to close this submission and invite a resubmission at a later point in time.

**Comment by csoneson** [comment]
@neuromusic - I'm going to pause this submission again. Unless I get a firm commitment to address the remaining issues within the next three weeks we will proceed with rejecting this submission and welcome a resubmission at a later stage.

**Comment by neuromusic** [comment]
Thanks @csoneson. We're working through fixing our CI to accommodate the requested changes. A pause to the submission sounds reasonable.

**Comment by csoneson** [comment]
👋 @neuromusic - just a gentle reminder that if we don't have a resolution of the few remaining issues here within a week, I will move to recommend rejecting the submission at its current stage and welcome a resubmission at a later point in time.

**Comment by neuromusic** [comment]
Thank you @csoneson! I have a PR open to address the outstanding issues. 2 of the 3 are fixed and I will have a solution for the final issue within the week (by Friday, 3/19).

https://github.com/spacetx/starfish/pull/1938

**Comment by csoneson** [comment]
Great, thanks @neuromusic! I will un-pause this submission then and check in again by the end of the week.

**Comment by csoneson** [comment]
@neuromusic - just a reminder that we're expecting an update here.

**Comment by neuromusic** [comment]
I've addressed the remaining issues. Thank you for your patience!

**Comment by csoneson** [comment]
Great, thank you! @vals - would you be able to take a look at the updates and see if they address your remaining concerns?

**Comment by csoneson** [comment]
@whedon generate pdf

**Comment by whedon** [comment]
:point_right::page_facing_up: [Download article proof](https://raw.githubusercontent.com/openjournals/joss-papers/joss.02440/joss.02440/10.21105.joss.02440.pdf) :page_facing_up: [View article proof on GitHub](https://github.com/openjournals/joss-papers/blob/joss.02440/joss.02440/10.21105.joss.02440.pdf) :page_facing_up: :point_left:

**Comment by vals** [comment]
I had a look at the changes, and it all looks fine now!

Best,
/Valentine

**Comment by csoneson** [comment]
Thank you @vals! So it looks like we're all good from the reviewers' side here. 
@neuromusic - I'll give it a final look through and get back to you shortly with the next steps.

**Comment by neuromusic** [comment]
Wonderful! Thank you all!

**Comment by csoneson** [comment]
👋 @neuromusic - I just sent a [PR](https://github.com/spacetx/starfish/pull/1942) with a couple of small fixes to the manuscript, mainly for the references formatting. Please check and merge if you agree. In addition, JOSS papers are currently [expected](https://joss.readthedocs.io/en/latest/submitting.html#what-should-my-paper-contain) to include an explicit `Statement of Need` section that illustrates the research purpose of the software. I think you could easily split off a part of your current `Summary` section to achieve this. 

Then, the next steps would be as follows: 
- [ ] Make a tagged release of your software, and list the version tag of the archived version here.
- [ ] Archive the reviewed software in Zenodo or a similar service
- [ ] Check that the Zenodo archive has the correct metadata. In particular, the title and author lists should match those of the paper.
- [ ] Please list the DOI of the archived version here.

I can then move on to accept the submission. Thanks!

**Comment by csoneson** [comment]
👋 @neuromusic - I just wanted to check in to see how things are going, and make sure that the next steps are clear. Once the things in the post above have been done, we can publish this paper. Let me know if you have any questions. Thanks!

**Comment by csoneson** [comment]
ping @neuromusic

**Comment by neuromusic** [comment]
thanks @csoneson, we'll cut a new release this week :D

**Comment by csoneson** [comment]
@neuromusic - could you update us on how things are going here? Please note that the archived version of the software should correspond closely to what the reviewers had a chance to look at - in other words, we don't expect there to be lots of new developments going on between the reviewers signing off and the software archive that will be linked with the JOSS publication being generated.

**Comment by csoneson** [comment]
@neuromusic - any news?

**Comment by neuromusic** [comment]
ok, finally cut the release & updated zenodo info.

there was no additional development besides fixing some dependencies for the release.

DOI here: 10.5281/zenodo.4734993

**Comment by neuromusic** [comment]
& zenodo link here: https://zenodo.org/record/4734993

**Comment by csoneson** [comment]
@whedon generate pdf

**Comment by whedon** [comment]
:point_right::page_facing_up: [Download article proof](https://raw.githubusercontent.com/openjournals/joss-papers/joss.02440/joss.02440/10.21105.joss.02440.pdf) :page_facing_up: [View article proof on GitHub](https://github.com/openjournals/joss-papers/blob/joss.02440/joss.02440/10.21105.joss.02440.pdf) :page_facing_up: :point_left:

**Comment by csoneson** [comment]
@whedon check references

**Comment by whedon** [comment]
```
Reference check summary (note 'MISSING' DOIs are suggestions that need verification):

OK DOIs

- 10.1016/j.cell.2019.05.031 is OK
- 10.1016/j.cell.2018.05.035 is OK
- 10.1038/nmeth.3252 is OK
- 10.1186/s13059-017-1382-0 is OK
- 10.1073/pnas.1612826113 is OK
- 10.1038/nmeth.2563 is OK
- 10.1038/s41592-018-0175-z is OK
- 10.1093/nar/gkx1206 is OK
- 10.1126/science.aat5691 is OK
- 10.5281/zenodo.59499 is OK
- 10.7717/peerj.453 is OK
- 10.5281/zenodo.3710410 is OK
- 10.1371/journal.pone.0228760 is OK
- 10.1093/nar/gkaa312 is OK

MISSING DOIs

- None

INVALID DOIs

- None
```

**Comment by csoneson** [comment]
@whedon set 10.5281/zenodo.4734993 as archive

**Comment by whedon** [comment]
OK. <a href="https://doi.org/10.5281/zenodo.4734993" target="_blank">10.5281/zenodo.4734993</a> is the archive.

**Comment by csoneson** [comment]
@whedon set 0.2.2 as version

**Comment by whedon** [comment]
OK. 0.2.2 is the version.

**Comment by csoneson** [comment]
@whedon accept

**Comment by whedon** [comment]
```
Attempting dry run of processing paper acceptance...
```

**Comment by csoneson** [comment]
Thanks @neuromusic! I'm handing this over now to the AEiC on rotation for the last checks.

**Comment by whedon** [comment]
```
Reference check summary (note 'MISSING' DOIs are suggestions that need verification):

OK DOIs

- 10.1016/j.cell.2019.05.031 is OK
- 10.1016/j.cell.2018.05.035 is OK
- 10.1038/nmeth.3252 is OK
- 10.1186/s13059-017-1382-0 is OK
- 10.1073/pnas.1612826113 is OK
- 10.1038/nmeth.2563 is OK
- 10.1038/s41592-018-0175-z is OK
- 10.1093/nar/gkx1206 is OK
- 10.1126/science.aat5691 is OK
- 10.5281/zenodo.59499 is OK
- 10.7717/peerj.453 is OK
- 10.5281/zenodo.3710410 is OK
- 10.1371/journal.pone.0228760 is OK
- 10.1093/nar/gkaa312 is OK

MISSING DOIs

- None

INVALID DOIs

- None
```

**Comment by whedon** [comment]
:wave: @openjournals/joss-eics, this paper is ready to be accepted and published.

 Check final proof :point_right: https://github.com/openjournals/joss-papers/pull/2276

If the paper PDF and Crossref deposit XML look good in https://github.com/openjournals/joss-papers/pull/2276, then you can now move forward with accepting the submission by compiling again with the flag `deposit=true` e.g.
 ```
@whedon accept deposit=true
```

**Comment by kthyng** [comment]
@csoneson You made this so easy! Thanks for sending this on so ready to go!! We can move this right on through the system since I see no errors or additional items to take care of.

**Comment by kthyng** [comment]
@whedon accept deposit=true

**Comment by whedon** [comment]
```
Doing it live! Attempting automated processing of paper acceptance...
```

**Comment by whedon** [comment]
🐦🐦🐦 👉 [Tweet for this paper](https://twitter.com/JOSS_TheOJ/status/1389584439106232322) 👈 🐦🐦🐦

**Comment by whedon** [comment]
🚨🚨🚨 **THIS IS NOT A DRILL, YOU HAVE JUST ACCEPTED A PAPER INTO JOSS!** 🚨🚨🚨

 Here's what you must now do:

0. Check final PDF and Crossref metadata that was deposited :point_right: https://github.com/openjournals/joss-papers/pull/2277
1. Wait a couple of minutes, then verify that the paper DOI resolves [https://doi.org/10.21105/joss.02440](https://doi.org/10.21105/joss.02440)
2. If everything looks good, then close this review issue.
3. Party like you just published a paper! 🎉🌈🦄💃👻🤘

 Any issues? Notify your editorial technical team...

**Comment by kthyng** [comment]
Congrats on your new publication @neuromusic! Thanks to editor @csoneson and reviewers @giovp, @shazanfar, and @vals for your time, expertise, and hard work!

**Comment by whedon** [comment]
:tada::tada::tada: Congratulations on your paper acceptance! :tada::tada::tada:

If you would like to include a link to your paper from your README use the following code snippets:

```
Markdown:
[![DOI](https://joss.theoj.org/papers/10.21105/joss.02440/status.svg)](https://doi.org/10.21105/joss.02440)

HTML:
<a style="border-width:0" href="https://doi.org/10.21105/joss.02440">
  <img src="https://joss.theoj.org/papers/10.21105/joss.02440/status.svg" alt="DOI badge" >
</a>

reStructuredText:
.. image:: https://joss.theoj.org/papers/10.21105/joss.02440/status.svg
   :target: https://doi.org/10.21105/joss.02440
```
This is how it will look in your documentation:

[![DOI](https://joss.theoj.org/papers/10.21105/joss.02440/status.svg)](https://doi.org/10.21105/joss.02440)

**We need your help!**

Journal of Open Source Software is a community-run journal and relies upon volunteer effort. If you'd like to support us please consider doing either one (or both) of the the following:

- Volunteering to review for us sometime in the future. You can add your name to the reviewer list here: https://joss.theoj.org/reviewer-signup.html
- Making a small donation to support our running costs here: https://numfocus.org/donate-to-joss
